frontend/src/LineGraph.res

```diff
@@ -320,7 +329,7 @@ let make = React.memo((
   let data = switch lastRow {
   | Some(lastRow) =>
     let lastIndex = DataRow.unsafe_get_index(lastRow)
-    BeltHelpers.Array.push(data, DataRow.nan2(~index=lastIndex + 1))
+    data->Belt.Array.sliceToEnd(0)->BeltHelpers.Array.push(DataRow.nan2(~index=lastIndex + 1))
```

Comment:
Also avoid mutation. Since we do this same logic in multiple places, we could also change the `BeltHelpers.Array.push` function to not mutate the array, and just use that one everywhere? And probably rename it to `add` or something else than `push`?

Reply:
Changing `BeltHelpers.Array.push` to not mutate and renaming is a good idea!